nikola/plugins/task/sources.py

```diff
@@ -61,12 +61,8 @@ def gen_tasks(self):
                     # do not publish PHP sources
                     if post.source_ext(True) == post.compiler.extension():
                         continue
-                    source = post.source_path
-                    if lang != kw["default_lang"]:
-                        source_lang = utils.get_translation_candidate(self.site.config, source, lang)
-                        if os.path.exists(source_lang):
-                            source = source_lang
-                    if os.path.isfile(source):
```

Comment:
This code essentially does what `post.translated_source_path(lang)` is supposed to be doing. Even worse, it uses a slightly different logic, which could cause incorrect source files to be copied.